docs/TypeDenotation.md

```diff
@@ -0,0 +1,55 @@
+Using Denotation For Semantic Description
+------------------------------------------
+
+Denotation is an experiment to give semantic description to models.  This enables model authors to describe the parts of their model application developers need to know in order to consume them.
+
+There are 2 types of denotation : [Type Denotation](TypeDenotation.md) and [Dimension Denotation](DimensionDenotation.md).
+
+### Type Denotation
+
+Type Denotation is used to describe semantic information around what the inputs and outputs are.    It is stored on the TypeProto message.
+
+#### Motivation
+
+The motivation of such a mechanism can be illustrated via a simple example. In the the neural network SqueezeNet, it takes in an NCHW image input float[1,2,244,244] and produces a output float[1,1000,1,1]:
+
+```
+input_in_NCHW -> data_0 -> SqueezeNet() -> output_softmaxout_1
+```
+
+In order to run this model the user needs a lot of information.    In this case the user needs to know:
```

Comment:
I think instead of using type denotation, saving the following information in the metadata is more straightforward.

Reply:
+1 multiple inputs is a good call